code/modules/mob/typing_indicator.dm

```diff
@@ -74,21 +81,35 @@ GLOBAL_LIST_EMPTY(typing_indicator)
 			if(length(temp) > 5 && findtext(temp, "Say \"", 1, 7))
 				set_typing_indicator(1)
 			else if(length(temp) > 3 && findtext(temp, "Me ", 1, 5))
-				set_typing_indicator(1)
+				set_typing_indicator(1, TRUE)
 
 			else
 				set_typing_indicator(0)
 
 /client/verb/typing_indicator()
 	set name = "Show/Hide Typing Indicator"
 	set category = "Preferences"
-	set desc = "Toggles showing an indicator when you are typing emote or say message."
+	set desc = "Toggles showing an indicator when you are typing a message."
 	prefs.toggles ^= PREFTOGGLE_SHOW_TYPING
 	prefs.save_preferences(src)
 	to_chat(src, "You will [(prefs.toggles & PREFTOGGLE_SHOW_TYPING) ? "no longer" : "now"] display a typing indicator.")
 
 	// Clear out any existing typing indicator.
 	if(prefs.toggles & PREFTOGGLE_SHOW_TYPING)
-		if(istype(mob)) mob.set_typing_indicator(0)
+		if(istype(mob))
+			mob.set_typing_indicator(0)
+
+	feedback_add_details("admin_verb", "TID") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+
+/client/verb/emote_indicator()
+	set name = "Show/Hide Emote Typing Indicator"
+	set category = "Preferences"
+	set desc = "Toggles showing an indicator when you are typing an emote."
+	prefs.toggles2 ^= PREFTOGGLE_2_EMOTE_BUBBLE
+	prefs.save_preferences(src)
+	to_chat(src, "You will [(prefs.toggles2 & PREFTOGGLE_2_EMOTE_BUBBLE) ? "no longer" : "now"] display a typing indicator for emotes.")
+
+	feedback_add_details("admin_verb", "EID")
 
-	feedback_add_details("admin_verb","TID") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+#undef TYPING_INDICATOR_LIFETIME
```

Comment:
Please add the comment that warns about copypaste to the feedback add details here.